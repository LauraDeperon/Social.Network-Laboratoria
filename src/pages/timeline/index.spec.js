import { Timeline } from './index.js';
import * as service from '../../services/index.js';
import * as util from '../../utils/history.js';

jest.mock('../../services/index.js');

util.onNavigate = jest.fn(() => Promise.resolve(true));

describe('Login', () => {
  it('should be a function', () => {
    expect(typeof Timeline).toBe('function');
  });
  it('should render the page', () => {
    expect(typeof Timeline).toMatchSnapshot();
  });
  it('should sign out the user', () => {
    Timeline().querySelector('#signout-button').dispatchEvent(new Event('click'));
    return expect(service.signOut).toHaveBeenCalled();
  });
  // it('should post a review', async () => {
  //   Timeline().querySelector('#publish-review').dispatchEvent(new Event('click'));
  //   await expect(service.UserProfileInfo).toHaveBeenCalled();
  //   expect(service.Review).toHaveBeenCalled();
  //   expect(util.onNavigate).toHaveBeenCalledWith('/timeline');
  // });
  it('should render the modal template', () => {
    expect(typeof modalTemplate).toMatchSnapshot();
  });
  // it('should edit and update a posted review', () => {
  //   Timeline().querySelector('#update-review').dispatchEvent(new Event('click'));
  //   return expect(service.ReviewPost).toHaveBeenCalled();
  // });
  it('should render the post template', () => {
    expect(typeof postTemplate).toMatchSnapshot();
  });
//   it('should delete a review', async () => {
//     Timeline().querySelectorAll('.delete-button').dispatchEvent(new Event('click'));
//     await expect(service.Review).toHaveBeenCalled();
//     expect(typeof Timeline).toMatchSnapshot();
//   });
});
