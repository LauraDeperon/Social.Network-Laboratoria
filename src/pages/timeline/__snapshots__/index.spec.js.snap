// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render the modal template 1`] = `"undefined"`;

exports[`Login should render the page 1`] = `"function"`;

exports[`Login should render the post template 1`] = `"undefined"`;
